lms/djangoapps/mobile_api/social_facebook/courses/tests.py

```diff
@@ -0,0 +1,267 @@
+        self.assertEqual(self.course.id._to_string().replace('+', '/'),
+                         response.data[0]['course']['id'])
+        self.assertEqual(self.course_2.id._to_string().replace('+', '/'),
+                         response.data[1]['course']['id'])
+
+    @httpretty.activate
+    def test_no_courses_with_friends_beacause_sharring_pref_off(self):
+        self.user_create_and_signin(1)
+        self.link_edX_account_to_social_backend(self.user_1, self.BACKEND, self.FB_ID_1)
+        self.set_sharing_preferences(self.user_1, False)
+        self.set_facebook_interceptor({'data':
+                                        [{'name': self.USERNAME_1,
+                                            'id': self.FB_ID_1}
+                                        ]})
+        self.enroll_in_course(self.user_1, self.course)
+        url = reverse('courses-with-friends')
+        response = self.client.get(url, {'oauth_token': self._FB_USER_ACCESS_TOKEN})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 0)
+
+    @httpretty.activate
+    def test_no_courses_with_friends_beacause_no_auth_token(self):
+        self.user_create_and_signin(1)
+        self.link_edX_account_to_social_backend(self.user_1, self.BACKEND, self.FB_ID_1)
+        self.set_sharing_preferences(self.user_1, False)
+        self.set_facebook_interceptor({'data':
+                                        [{'name': self.USERNAME_1,
+                                            'id': self.FB_ID_1}
+                                        ]})
+        self.enroll_in_course(self.user_1, self.course)
+        url = reverse('courses-with-friends')
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 400)
+
+    def test_no_courses_with_friends_beacause_bad_auth_token(self):
+        self.user_create_and_signin(1)
+        self.link_edX_account_to_social_backend(self.user_1, self.BACKEND, self.FB_ID_1)
+        self.set_sharing_preferences(self.user_1, False)
+        self.enroll_in_course(self.user_1, self.course)
+        url = reverse('courses-with-friends')
+        response = self.client.get(url, {'oauth_token': self._FB_USER_ACCESS_TOKEN + "this_is_a_bad_token"})
+        self.assertEqual(response.status_code, 400)
+
+
+    '''
+        Helper Functions 
+    '''
+
+    def set_sharing_preferences(self, user, boolean_value):
+        ''' Sets self.user's share settings to True
+        '''
+        update_preferences(user.username, share_pref=boolean_value)
+        self.assertEqual(preference_info(user.username)['share_pref'], unicode(boolean_value))
+
+
+    def set_facebook_interceptor(self, data):
+        httpretty.register_uri(httpretty.GET,
+                        "https://graph.facebook.com/v2.2/me/friends",
+                        body=json.dumps(data),
+                        status=201)
+
+    def user_create_and_signin(self, user_number):
+        if user_number == 1:
```

Comment:
Nit: If the users were listed in an array, wouldn't need to duplicate the logic for each one.